Generated with 'git diff --no-prefix > tensorflow.patch'

diff --git BUILD BUILD
index 4bf647e47a..b1a4ecbac2 100644
Exports additional files. Fixed in v1.13.
--- BUILD
+++ BUILD
@@ -2,5 +2,8 @@ exports_files(
     [
         "LICENSE",
         "ACKNOWLEDGEMENTS",
+        "configure",
+        "configure.py",
+        "tools/bazel.rc",
     ],
 )

diff --git tensorflow/BUILD tensorflow/BUILD
index b5e0a4e98b..c7e264f720 100644
Make dependencies public. Fixed in v1.12.
--- tensorflow/BUILD
+++ tensorflow/BUILD
@@ -553,6 +553,7 @@ tf_cc_shared_object(
         "//tensorflow/c/eager:c_api",
         "//tensorflow/core:tensorflow",
     ],
+    visibility = ["//visibility:public"],
 )

 tf_cc_shared_object(
@@ -580,6 +581,7 @@ tf_cc_shared_object(
         "//tensorflow/cc/profiler",
         "//tensorflow/core:tensorflow",
     ] + if_ngraph(["@ngraph_tf//:ngraph_tf"]),
+    visibility = ["//visibility:public"],
 )

 exports_files(

diff --git third_party/gpus/cuda_configure.bzl third_party/gpus/cuda_configure.bzl
index f6a39aeaf1..a4ee4f12a1 100644
Remove Windows specific code that would require bazel v0.13.
--- third_party/gpus/cuda_configure.bzl
+++ third_party/gpus/cuda_configure.bzl
@@ -120,7 +120,6 @@ load(
     "@bazel_tools//tools/cpp:windows_cc_configure.bzl",
     "find_msvc_tool",
     "find_vc_path",
-    "setup_vc_env_vars",
 )
 
 def _get_python_bin(repository_ctx):
@@ -174,7 +173,7 @@ def _get_win_cuda_defines(repository_ctx):
                             "Please check your installation following https://docs.bazel.build/versions/master/windows.html#using")
         return {}

-    env = setup_vc_env_vars(repository_ctx, vc_path)
+    env = {}
     escaped_paths = escape_string(env["PATH"])
     escaped_include_paths = escape_string(env["INCLUDE"])
     escaped_lib_paths = escape_string(env["LIB"])

diff --git tensorflow/workspace.bzl tensorflow/workspace.bzl
index 758c94c542..1366249d32 100755
Use abseil version which includes SwissTable.
--- tensorflow/workspace.bzl
+++ tensorflow/workspace.bzl
@@ -106,11 +106,11 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
     tf_http_archive(
         name = "com_google_absl",
         urls = [
-            "https://mirror.bazel.build/github.com/abseil/abseil-cpp/archive/f0f15c2778b0e4959244dd25e63f445a455870f5.tar.gz",
-            "https://github.com/abseil/abseil-cpp/archive/f0f15c2778b0e4959244dd25e63f445a455870f5.tar.gz",
+            "https://mirror.bazel.build/github.com/abseil/abseil-cpp/archive/48cd2c3f351ff188bc85684b84a91b6e6d17d896.tar.gz",
+            "https://github.com/abseil/abseil-cpp/archive/48cd2c3f351ff188bc85684b84a91b6e6d17d896.tar.gz",
         ],
-        sha256 = "4ee36dacb75846eaa209ce8060bb269a42b7b3903612ca6d9e86a692659fe8c1",
-        strip_prefix = "abseil-cpp-f0f15c2778b0e4959244dd25e63f445a455870f5",
+        sha256 = "7dd09690ae7ca4551de3111d4a86b75b23ec17445f273d3c42bdcdc1c7b02e4e",
+        strip_prefix = "abseil-cpp-48cd2c3f351ff188bc85684b84a91b6e6d17d896",
         build_file = clean_dep("//third_party:com_google_absl.BUILD"),
     )

diff --git tensorflow/tensorflow.bzl tensorflow/tensorflow.bzl
index adac895a17..b768c38a35 100644
--- tensorflow/tensorflow.bzl
+++ tensorflow/tensorflow.bzl
@@ -1359,9 +1359,9 @@ _transitive_hdrs = rule(
     implementation = _transitive_hdrs_impl,
 )
 
-def transitive_hdrs(name, deps = [], **kwargs):
-    _transitive_hdrs(name = name + "_gather", deps = deps)
-    native.filegroup(name = name, srcs = [":" + name + "_gather"])
+def transitive_hdrs(name, deps = [], tags = [], **kwargs):
+    _transitive_hdrs(name = name + "_gather", tags = tags, deps = deps)
+    native.filegroup(name = name, srcs = [":" + name + "_gather"], tags = tags)
 
 # Create a header only library that includes all the headers exported by
 # the libraries in deps.
